ARCHITECTURE.md

```diff
@@ -0,0 +1,133 @@
+| nginx                   | Receive all API requests and route them appropriately.                                                                                                  |
+| Edge Endpoint           | Receive inference requests and determine whether to handle them locally or send them to the cloud.                                                      |
+| Inference Model Updater | Keep track of which models are in use, download model data, and start pods to serve inference on those models, updating to the latest models regularly. |
+| Status Monitor          | Aggregate usage stats and upload them to the cloud periodically.                                                                                        |
+
+## Network flow
+
+By default, the edge endpoint exposes the Groundlight API on port 30101 on the local machine.
+
+The following diagram shows how HTTP requests are handled by the edge endpoint. 
+
+<img src="images/Client request processing.excalidraw.png" alt="Client request processing" width="800"/>
+
+(The inference model updater does not handle any requests. See below in [Communication between the edge endpoint containers](#communication-between-the-edge-endpoint-containers) for more details.)
+
+How URLs are handled:
+
+| URL                               | Verb(s) | Handled by                           |
+| --------------------------------- | ------- | ------------------------------------ |
+| `/image-queries`                  | `POST`  | Edge endpoint, may escalate to cloud |
+| `/health/live`<br>`/health/ready` | `GET`   | Edge endpoint                        |
+| `/ping`                           | `GET`   | Edge endpoint                        |
+| `/status`                         | `GET`   | Status monitor                       |
+| all others                        | all     | Forward to cloud                     |
+
+The nginx server will always try to send API calls to the the edge endpoint first. If the edge endpoint cannot handle the request (i.e., it returns HTTP error 404 `Page not found`, 405 `Method not allowed`, or 422 `Unprocessable Content`), nginx will use its fallback mechanism to forward the call to the cloud service. 
+
+The edge endpoint container handles all inference requests (`POST /image-queries`) and will either send them to a local inference pod or escalate them to the cloud directly (see [Inference Requests](#inference-requests)) below for more details.
+
+The following sequence diagrams illustrate the process of handling requests:
+
+#### Local inference 
+```mermaid
+sequenceDiagram
+actor client
+box Edge endpoint pod
+participant nginx
+participant edge-endpoint
+end
+participant inference pod
+participant cloud service
+client->>nginx: POST /image-queries
+nginx->>edge-endpoint: POST /image-queries
+edge-endpoint->>+inference pod: POST /infer
+inference pod-->>-edge-endpoint: High-confidence result
+edge-endpoint-->>nginx: result
+nginx-->>client: result
+```
```

Comment:
TIL!

Reply:
Yeah, mermaid is awesome for throwing stuff into READMEs and such.